pulp_container/app/viewsets.py

```diff
@@ -696,6 +696,7 @@ class ContainerRepositoryViewSet(
                 "condition": [
                     "has_model_or_obj_perms:container.build_image_containerrepository",
                     "has_model_or_obj_perms:container.view_containerrepository",
+                    "has_repository_model_or_obj_perms:file.view_filerepository",
```

Comment:
Due to this line, I am receiving an HTTP 500 error.
```
[pulp]  | pulp [d8943394f02c4759b3deac8f518d2218]: django.request:ERROR: Internal Server Error: /pulp/api/v3/repositories/container/container/01907884-8984-7905-bc76-58b8c747c347/build_image/
[pulp]  | Traceback (most recent call last):
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
[pulp]  |     response = get_response(request)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
[pulp]  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
[pulp]  |     return view_func(*args, **kwargs)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 124, in view
[pulp]  |     return self.dispatch(request, *args, **kwargs)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
[pulp]  |     response = self.handle_exception(exc)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
[pulp]  |     self.raise_uncaught_exception(exc)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
[pulp]  |     raise exc
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 497, in dispatch
[pulp]  |     self.initial(request, *args, **kwargs)
[pulp]  |   File "/src/pulpcore/pulpcore/app/viewsets/base.py", line 300, in initial
[pulp]  |     super().initial(request, *args, **kwargs)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 415, in initial
[pulp]  |     self.check_permissions(request)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 332, in check_permissions
[pulp]  |     if not permission.has_permission(request, self):
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_access_policy/access_policy.py", line 69, in has_permission
[pulp]  |     allowed = self._evaluate_statements(statements, request, view, action)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_access_policy/access_policy.py", line 113, in _evaluate_statements
[pulp]  |     matched = self._get_statements_matching_conditions(
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_access_policy/access_policy.py", line 262, in _get_statements_matching_conditions
[pulp]  |     passed = self._check_condition(condition, request, view, action)
[pulp]  |   File "/usr/local/lib/python3.9/site-packages/rest_access_policy/access_policy.py", line 286, in _check_condition
[pulp]  |     result = method(request, view, action, arg)
[pulp]  |   File "/src/pulpcore/pulpcore/app/global_access_conditions.py", line 407, in has_repository_model_or_obj_perms
[pulp]  |     return request.user.has_perm(permission) or has_repository_obj_perms(
[pulp]  |   File "/src/pulpcore/pulpcore/app/global_access_conditions.py", line 367, in has_repository_obj_perms
[pulp]  |     plugin_repository = Repository.objects.get(pk=view.kwargs["repository_pk"]).cast()
[pulp]  | KeyError: 'repository_pk'
```

Reproducer:
```
pulp user create --username test --password krokodyl1 --email test@test.com
pulp user role-assignment add --username test --role "container.containerdistribution_collaborator" --object ""
pulp user role-assignment add --username test --role "container.containerrepository_content_manager" --object ""

FILE_REPO=$(pulp file repository create --name foo --autopublish | jq -r '.pulp_href')
pulp file content upload --repository foo --file ./setup.py --relative-path setup.py

CONTAINER_REPO=$(pulp container repository create --name build | jq -r '.pulp_href') 

http -a test:krokodyl1 --form POST :5001${CONTAINER_REPO}build_image/ "containerfile@./Containerfile" repo_version=${FILE_REPO}versions/1/
```

-------------------------------------

I do not think we can inject the policy from other entities' contexts here (repo_version=FILE_HREF). The policy is available only for the current viewset scope (CONTAINER_REPO). I think we need to run permission checks manually. Similarly to https://github.com/pulp/pulp_container/blob/a832b9fc9a06cf96fda20fc7203f06a2baa6b9c4/pulp_container/app/registry_api.py#L531 or create a customized policy condition, like in https://github.com/pulp/pulp_container/blob/a832b9fc9a06cf96fda20fc7203f06a2baa6b9c4/pulp_container/app/global_access_conditions.py#L55.

@mdellweg, any ideas on how to properly add permission checks for entities created from within a different plugin? In our case, we want to check if a user has permissions to read pulp_file repository and build an image inside pulp_container repository.